src/Plugins/RpcServer/Model/SignerOrWitness.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// SignerOrWitness.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Cryptography.ECC;
+using Neo.Json;
+using Neo.Network.P2P.Payloads;
+using Neo.Wallets;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Neo.Plugins.RpcServer.Model;
+
+public class SignerOrWitness
```

Comment:
Why not make a `base` class for common stuff? Than two different classes one, being `Signer` and other being `Witness`. This way we can get `Signer` attributes like `Rules` or other properties that separate the two classes.

Reply:
make it work first, optimize it later. I think you are better than me to make the code more elegent.